src/testing/react/__tests__/MockedProvider.test.tsx

```diff
@@ -743,35 +743,50 @@ describe("General use", () => {
     ).toBeInTheDocument();
   });
 
-  it("should support loading state testing with infinite delay", async () => {
-    function Component({ username }: Variables) {
-      const { loading, data } = useQuery<Data, Variables>(query, { variables });
+  it.each([Infinity, 30000])(
+    "should support loading state testing with %s delay",
+    async (timeout) => {
+      function Component({ username }: Variables) {
+        const { loading, data } = useQuery<Data, Variables>(query, {
+          variables,
+        });
 
-      if (loading || data === undefined) return <p>Loading the user ID...</p>;
+        if (loading) return <p>Loading the user ID...</p>;
+        if (data === undefined) return <p>Undefined data</p>;
 
-      return <p>The user ID is '{data.user.id}'</p>;
-    }
+        return <p>The user ID is '{data.user.id}'</p>;
+      }
 
-    const mocks: ReadonlyArray<MockedResponse> = [
-      {
-        delay: Infinity, // keep loading forever.
-        request: {
-          query,
-          variables,
+      const mocks: ReadonlyArray<MockedResponse> = [
+        {
+          delay: timeout, // keep loading forever.
+          request: {
+            query,
+            variables,
+          },
+          result: { data: { user } },
         },
-      },
-    ];
+      ];
 
-    render(
-      <MockedProvider mocks={mocks}>
-        <Component {...variables} />
-      </MockedProvider>
-    );
+      render(
+        <MockedProvider mocks={mocks}>
+          <Component {...variables} />
+        </MockedProvider>
+      );
 
-    expect(
-      await screen.findByText("Loading the user ID...")
-    ).toBeInTheDocument();
-  });
+      expect(
+        await screen.findByText("Loading the user ID...")
+      ).toBeInTheDocument();
+
+      wait(5000);
```

Comment:
```suggestion
      await wait(5000);
```

You'll probably want an `await` here to ensure this test actually waits the full 5 seconds before checking the loading state again.

That being said, this changes causes a couple new issues.

1. Jest will start failing these tests due to test timeouts (by default Jest tests are set to 5 seconds before timing out)
2. If you increase the threshold, these tests take at least 5 seconds to execute, which adds quite a bit of overhead.

I'd suggest using a [fake timer](https://jestjs.io/docs/timer-mocks) here instead to reduce the amount of time this test takes to execute. This also means you can increase the `wait` here to as long as you want. Something closer to 30 sec might be nicer to give a bit more confidence that our implementation is working correctly.

Reply:
Good call! I'll use fake timers here.

Reply:
Updated in [7f0a260](https://github.com/apollographql/apollo-client/pull/11472/commits/7f0a26098c05c32b26606ad66b3b8aef5ba21d4f)